### PR TITLE
InSty improvements

### PR DESCRIPTION
**Pull request by karolamik13**
I fixed a problem with calcStatisticsInteractions(), added some info warnings for InSty tool and WatFinder.
I also added a new function showInteractionsHist(), which generates bar plots with interactions (see below).

sbs_barplot
 
Now we have two options - bar plot and graph like the one below.

sbs_iPla2B

### CODE REVIEW

File: prody/proteins/interactions.py [hunk 1]
```diff
@@ -211,12 +211,15 @@ def get_energy(pair, source):
...
+        with pkg_resources.path('prody.proteins', 'tabulated_energies.txt') as file_path:
+            #print("File with energies: ", file_path) 
+            with open(file_path) as f:
+                data = np.loadtxt(f, dtype=str)
     else:
         file_path = pkg_resources.resource_filename('prody.proteins', 'tabulated_energies.txt')
-
-    with open(file_path) as f:
-        data = np.loadtxt(f, dtype=str)
+        #print("File with energies: ", file_path)
```
Comment (jamesmkrieger) [inline prody/proteins/interactions.py:220]: the commented line can be deleted

File: prody/proteins/interactions.py [hunk 2]
```diff
@@ -211,12 +211,15 @@ def get_energy(pair, source):
...
+            #print("File with energies: ", file_path) 
+            with open(file_path) as f:
+                data = np.loadtxt(f, dtype=str)
     else:
         file_path = pkg_resources.resource_filename('prody.proteins', 'tabulated_energies.txt')
-
-    with open(file_path) as f:
-        data = np.loadtxt(f, dtype=str)
+        #print("File with energies: ", file_path)  
+        with open(file_path) as f:
```
Comment (jamesmkrieger) [inline prody/proteins/interactions.py:221]: these makes sense

File: prody/proteins/interactions.py [hunk 3]
```diff
@@ -2102,7 +2171,7 @@ def calcStatisticsInteractions(data, **kwargs):
                     "energy": get_energy([element.split('-')[0][:3], element.split('-')[1][:3]], energy_list_type)
                 }
             except:
-                LOGGER.warn('energy information is not available for ', element.split('-')[0][:3], element.split('-')[1][:3])
+                LOGGER.warn('energy information is not available for ', element)
```
Comment (jamesmkrieger) [inline prody/proteins/interactions.py:2174]: I'll trust this works

File: prody/proteins/interactions.py [hunk 4]
```diff
@@ -2124,6 +2193,10 @@ def calcStatisticsInteractions(data, **kwargs):
         else: pass
     
     statistic.sort(key=lambda x: x[1], reverse=True)
+    
+    if statistic == []:
+        LOGGER.info("Decrease the value of weight_cutoff parameter to obtain the results. Default is 0.2.")
```
Comment (jamesmkrieger) [inline prody/proteins/interactions.py:2198]: This could be a bit clearer. Perhaps say that results were filtered out or whatever it is that's happening.

**Review by jamesmkrieger — APPROVED** [review]
seems good. I'd just suggest improving the logging when statistic is empty like the error for no waters

File: prody/proteins/interactions.py (hunk 4 above)
Comment (jamesmkrieger) [reply prody/proteins/interactions.py:2198]: Makes sense but too long for message. How about "No meaningful interactions found. Decrease weight_cutoff to obtain some results (default is 0.2)."?

Comment (karolamik13) [reply prody/proteins/interactions.py]: I replaced it.

Comment (jamesmkrieger) [reply prody/proteins/interactions.py]: Looks good. Ready to merge

File: prody/proteins/interactions.py (hunk 3 above)
Comment (jamesmkrieger) [reply prody/proteins/interactions.py:2172]: Ok, I agree it’s helpful to have all the details

**Pull request merged**